### PR TITLE
Fix the bug about extra parentheses around symbolic names (issue #273)

### PR DESCRIPTION
**Pull request by sighingnow**
Hindent use `prettyPrint` from the haskell-src-exts to pretty print a `Name`, see [L1217](https://github.com/chrisdone/hindent/blob/master/src/HIndent/Pretty.hs#L1217), but when print a symbolic name separately, extra parentheses will be added.

~~~haskell
 > prettyPrint $ Symbol undefined "++"
"(++)"
 > prettyPrint $ Symbol undefined "L.++"
"(L.++)"
~~~

But when pretty print a whole expression, it works as expected.

~~~haskell
 > prettyPrint $ Symbol undefined "xs ++ ys"
"xs ++ ys"
 > prettyPrint $ Symbol undefined "xs L.++ ys"
"xs (L.++) ys"
~~~

This patch fixes issue #273 and also part of issue #277 (it was fixed by f49fb09d). I'm not sure whether its a bug of haskell-src-exts, or just a feature.

### CODE REVIEW

**Comment by chrisdone** [comment]
Can you add a regression test in TESTS.md?

**Comment by sighingnow** [comment]
There's already a regression test case for issue 277 in TEST.md. I think it's enough to address this bug about formatting symbolic names wrongly.

File: TESTS.md [hunk 1]
```diff
@@ -852,6 +852,18 @@ data Link c1 c2 a c =
...
 
+ttuegel qualified infix sections get mangled #273
+
+```haskell
+-- https://github.com/chrisdone/hindent/issues/273
+import qualified Data.Vector as V
+
+main :: IO ()
+main = do
+  let _ = foldr1 V.++ [V.empty, V.empty]
```
Comment (chrisdone) [inline TESTS.md]: I think this is supposed to be `(V.++)`. This is basically `x ++ y`. Can you fix it?

Comment (chrisdone) [inline TESTS.md]: See here: https://github.com/chrisdone/hindent/issues/273

Comment (sighingnow) [inline TESTS.md]: I have fixed it. It was caused by the wrong pretty printing for `Var` in `exp` function. I also have added there more corner cases to regression test.

**Comment by chrisdone** [comment]
Great!
